dissect/target/plugins/os/unix/linux/android/_os.py

```diff
@@ -15,24 +16,36 @@ def __init__(self, target: Target):
         super().__init__(target)
         self.target = target
 
+        self.build_prop_paths = set(self._find_build_props(self.target.fs))
         self.props = {}
-        if (build_prop := self.target.fs.path("/build.prop")).exists():
-            self.props = configutil.parse(build_prop, separator=("=",), comment_prefixes=("#",)).parsed_data
+
+        for build_prop in self.build_prop_paths:
+            try:
+                self.props.update(configutil.parse(build_prop, separator=("=",), comment_prefixes=("#",)).parsed_data)
+
+            except Exception as e:
+                self.target.log.warning("Unable to parse Android build.prop file %s: %s", build_prop, e)
+                pass
+
+    @staticmethod
+    def _find_build_props(fs: Filesystem) -> Iterator[Path]:
+        for prop in fs.path("/").glob("**/build.prop"):
```

Comment:
Is this not super expensive? Won't this iterate over every directory on every filesystem on every Linux-like target?